src/phpDocumentor/Descriptor/Argument.php

```diff
@@ -13,58 +13,74 @@
 /**
  * Descriptor representing a single Argument of a method or function.
  */
-class ArgumentDescriptor extends DescriptorAbstract implements Interfaces\ArgumentInterface
+final class Argument
 {
+    /**
+     * @var string name of the Argument
+     */
+    private $name = null;
+
     /** @var string[] $type an array of normalized types that should be in this Argument */
-    protected $types = array();
+    private $types = array();
 
     /** @var string|null $default the default value for an argument or null if none is provided */
-    protected $default;
+    private $default = null;
 
     /** @var bool $byReference whether the argument passes the parameter by reference instead of by value */
-    protected $byReference = false;
+    private $byReference = false;
 
     /** @var boolean Determines if this Argument represents a variadic argument */
-    protected $isVariadic = false;
+    private $isVariadic = false;
 
     /**
-     * {@inheritDoc}
+     * Initializes the object.
+     *
+     * @param string $name
+     * @param string $default
+     * @param bool $byReference
+     * @param bool $isVariadic
      */
-    public function setTypes($types)
+    public function __construct($name, $default = null, $byReference = false, $isVariadic = false)
     {
-        $this->types = $types;
+        $this->name = $name;
+        $this->default = $default;
+        $this->byReference = $byReference;
+        $this->isVariadic = $isVariadic;
     }
 
     /**
-     * {@inheritDoc}
+     * Returns the name of this argument.
+     *
+     * @return string
      */
-    public function getTypes()
+    public function getName()
     {
-        return $this->types;
+        return $this->name;
     }
 
     /**
      * {@inheritDoc}
      */
-    public function setDefault($value)
+    public function getTypes()
     {
-        $this->default = $value;
+        return $this->types;
     }
 
     /**
-     * {@inheritDoc}
+     * Add a type.
+     * @param string $type
      */
-    public function getDefault()
+    public function addType($type)
```

Comment:
FYI: In the ReflectionDocBlock component types will become classes and we can change the plural to a singular type since `string|int` will become a single class that represents a composite/mixed type containing both string and int. (I have no name for that class yet; it is not properly named in the Ubiquitous Language yet)